fuzz/fuzz_targets/compare_xz.rs

```diff
@@ -0,0 +1,36 @@
+#![no_main]
+#[macro_use]
+extern crate libfuzzer_sys;
+extern crate lzma_rs;
+extern crate xz2;
+
+use std::io::Read;
+use lzma_rs::error::Result;
+use xz2::stream;
+
+fn decode_xz_lzmars(compressed: &[u8]) -> Result<Vec<u8>> {
+    let mut bf = std::io::Cursor::new(compressed);
+    let mut decomp: Vec<u8> = Vec::new();
+    lzma_rs::xz_decompress(&mut bf, &mut decomp)?;
+    Ok(decomp)
+}
+
+fn decode_xz_xz2(compressed: &[u8]) -> Result<Vec<u8>> {
+    let bf = std::io::Cursor::new(compressed);
+    let mut decomp: Vec<u8> = Vec::new();
+    // create new XZ decompression stream with 8Gb memory limit and checksum verification disabled
+    let xz_stream = stream::Stream::new_stream_decoder(8*1024*1024*1024, stream::IGNORE_CHECK).expect("Failed to create stream");
```

Comment:
I'm surprised that the `cargo fmt -- --check` didn't trigger.
I'm even more surprised that Travis-CI didn't report any test results on the pull request.